src/DynamoPublishTests/PublishModelTests.cs

```diff
@@ -50,5 +51,22 @@ public void SendWorkspaceTest()
             Assert.IsNotNull(publishModel.HomeWorkspace);
             Assert.AreEqual(1, publishModel.CustomNodeWorkspaces.Count);
         }
+
+        [Test]
+        public void PublishModelStateTest()
+        {
+            string openPath = Path.Combine(TestDirectory, @"Libraries\DynamoPublishTests\PublishWorkspaceTest.dyn");
+            OpenModel(openPath);
+
+            publishModel.SendAsynchronously(CurrentDynamoModel.Workspaces);
+            Assert.AreEqual(PublishModel.UploadState.Uploading, publishModel.State);
+
+            // Wait until workspace is sent.
+            while (publishModel.State == PublishModel.UploadState.Uploading)
```

Comment:
I'd recommend adding some certain of termination condition. For example, increment a counter and if the counter goes over a certain threshold - `Assert.Fail`.  It's not acceptable for a test to hang indefinitely if the behavior is incorrect.